routes/messages.js

```diff
@@ -0,0 +1,89 @@
+const express = require('express');
+const router = express.Router();
+const messages = require('../Messages')
+
+
+
+
+// - [ ] Read all messages
+
+router.get("/", (request, response) => {
+    response.json(messages)
+});
+
+// - [ ] Read all messages
+
+router.get("/latest", (request, response) => {
+    response.json(messages.slice(-10))
+});
+
+
+
+//Search in messages
+
+router.get("/search", (request, response) => {
+    const queryWord = request.query.text
+    if (queryWord) {
+        const res = messages.filter(message => message.text.toLowerCase().includes(queryWord.toLowerCase()))
+        response.json(res)
+    }
+})
+
+
+// - [ ] Read one message specified by an ID
+router.get("/:id", (request, response) => {
+    const id = request.params.id
+    const res = messages.find(message => message.id === parseInt(id))
+    res ? response.json(res) : response.status(404).json({ msg: 'Message not found' })
+
+});
+
+// - [ ] Create a new message
+
+router.post("/", (request, response) => {
+    let time = new Date()
+    const newMessage = {
```

Comment:
More validation is needed in general

Reply:
Try to get your validation before you create any object or do any memory storage